apis/gitlab.go

```diff
@@ -21,6 +21,7 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 
 	projectsv1alpha1 "github.com/crossplane-contrib/provider-gitlab/apis/projects/v1alpha1"
+	v1alpha1 "github.com/crossplane-contrib/provider-gitlab/apis/v1alpha1"
```

Comment:
I started with that but changed it because deploy keys is a [top level api resource](https://docs.gitlab.com/ee/api/deploy_keys.html) on gitlab (e.g. `https://gitlab.example.com/api/v4/deploy_keys`)

If I put it in a subgroup, the yaml representation would look a bit weird too:
```yaml
apiVersion: deploykeys.gitlab.crossplane.io/v1alpha1
kind: DeployKey
```

Reply:
I don't see an issue with

```yaml
apiVersion: deploykeys.gitlab.crossplane.io/v1alpha1
kind: DeployKey
```

What exactly is weird about it?

Reply:
probably just nitpicking here, but `kind` has the same name as its parent `group`. Thus `group` seems a bit duplicated, given that there is just a single `kind` in it?

Reply:
`deploykeys.gitlab.crossplane.io` is just a package namespace. `DeployKey` is a component in the `deploykeys` API so it makes sense the put it there.

The root `gitlab.crossplane.io` API should be reserved to provider-specific resources, such as `ProviderConfig`.

There are also single-resource APIs in other providers, such as `secretsmanager` in `provider-aws`.

Reply:
> deploykeys.gitlab.crossplane.io is just a package namespace. DeployKey is a component in the deploykeys API so it makes sense the put it there.

Yeah, guess the naming is just a bit odd. Will change back 👍